### PR TITLE
fix SM100 FP8 accuracy issue

### PR DESCRIPTION
**Pull request by chenqianfzh**

### CODE REVIEW

File: python/sglang/srt/layers/moe/moe_runner/asym_gemm.py [hunk 1]
```diff
@@ -218,6 +218,46 @@ def _cast_to_e8m0_with_rounding_up(x: torch.Tensor) -> torch.Tensor:
...
+    exp = torch.bitwise_right_shift(temp, 23)
+    mant = torch.bitwise_and(temp, 0x7FFFFF)
+    is_ru = torch.logical_and(
+        torch.logical_and((mant > 0), (exp != 0xFE)),
+        ~torch.logical_and((exp == 0), (mant <= 0x400000)),
+    )
+    exp_rounded = torch.where(is_ru, exp + 1, exp)
+    ue8m0_fp32 = torch.bitwise_left_shift(exp_rounded, 23).view(torch.float32)
+
+    ratio = fp32_scale / ue8m0_fp32
```
[the remark this answers is not in the document]
Comment (chenqianfzh) [reply python/sglang/srt/layers/moe/moe_runner/asym_gemm.py:244]: This is actually safe in practice — per_token_group_quant_fp8 computes scales with eps=1e-10, so the scale is never exactly zero, even for all-zero activation blocks. A scale of 1e-10 rounds up to a valid power of 2, making the division well-defined.
